### ISSUE TITLE
[Bug] DSC-Loss with Batch-size greater one results in DSC values > 1

### ISSUE THREAD

**Issue by LeHenschel** [Deep-MI/3d-neuro-seg#1]
The current implementation leads to DSC loss with values above 1 if the batch-size is greater than one.

Problem is this part: https://github.com/Deep-MI/3d-neuro-seg/blob/main/utils/losses.py#L46 

The sum over the batch dimension should either be done earlier (before dividing numerator and denominator and subtracting from one in L44) or should be changed to average across the batches.

Correction:
```
# First dimensions H x W x D, then batch = dim 0, just leave channel = dim 1
numerator = 2 * intersection.sum(tuple(range(2, intersection.ndim))).sum(0)
denominator = union.sum(tuple(range(2, union.ndim))).sum(0) + eps
```

Minimal example (2D for simplicity):
```
from losses import DiceLoss

# Generate random one-hot encoded prediction
output = torch.ones((2, 3, 3, 3))
output[:, 0, ...] = 0
output[:, 2, ...] = 0

# Corresponding ground truth
target = torch.ones((2, 1, 3, 3))
target[1, :, 1:2, :] = 2
target[0, :, 1:3, :] = 2
target[:, :, 1, 0] = 0
target[:, :, 0, 2] = 0
target = target.type(torch.LongTensor)

loss = DiceLoss().forward(output, target)
# Result: tensor(1.6407), should be tensor(0.8133)
```

**Comment by dkuegler** [comment]
This is indeed incorrect, if we are being strict.
Since the batch size is a constant (and in our case 1), the effect for the training is very limited though.

If you have a PR, we are happy to merge this, as it does not break reproducibility (for batch size 1 as in our experiments).
